nupic/regions/TPRegion.py

```diff
@@ -506,6 +517,17 @@ def _compute(self, inputs, outputs):
       size = activeLearnCells.shape[0] * activeLearnCells.shape[1]
       outputs['lrnActiveStateT'][:] = activeLearnCells.reshape(size)
 
+    if self.computePredictedActiveCellIndices:
+      # Reshape so we are dealing with 1D arrays
+      activeState = self._tfdr.getActiveState().reshape(-1).astype('float32')
+      predictedState = self._tfdr.getPredictedState().reshape(-1).astype('float32')
+      activeIndices = set(numpy.where(activeState != 0)[0])
+      predictedIndices= set(numpy.where(predictedState != 0)[0])
+      predictedActiveIndices = activeIndices & predictedIndices
```

Comment:
Instead of converting to sets and using the `&` operator, use `numpy.intersect1d`.

Plus then `activeIndices` and `predictedIndices` are numpy arrays which can be used as indices below without needed to convert to a Python list.